config/settings.rb

```diff
@@ -11,6 +12,7 @@
 SETTINGS[:puppetconfdir] ||= '/etc/puppet'
 SETTINGS[:puppetvardir]  ||= '/var/lib/puppet'
 SETTINGS[:puppetssldir]  ||= "#{SETTINGS[:puppetvardir]}/ssl"
+SETTINGS[:domain] ||= Facter.value(:domain) || Facter.value(:hostname)
```

Comment:
I don't think it's meant to work when the domain is set to the hostname. It certainly avoids the error though...